### PR TITLE
Logistic growth

### PR DESCRIPTION
**Pull request by Riley16**

### CODE REVIEW

File: neuralprophet/time_net.py [hunk 1]
```diff
@@ -302,6 +335,99 @@ def _piecewise_linear_trend(self, t):
...
+        k_t = self.trend_k0 * torch.ones_like(t)
+        m_t = self.trend_m0 * torch.ones_like(t)
+        for s, t_s in enumerate(self.trend_changepoints_t):
+            indx = t >= t_s
+            k_t[indx] += self.trend_deltas[s]
+            m_t[indx] += gammas[s]
+
+        return (self.trend_cap - self.trend_floor) / (1 + torch.exp(-k_t * (t - m_t))) + self.trend_floor
+
+    def init_logistic_growth(self, dataset, data_params):
```
Comment (ourownstory) [inline neuralprophet/time_net.py:368]: let's run this init in a helper function

File: neuralprophet/time_net.py [hunk 2]
```diff
@@ -302,6 +335,99 @@ def _piecewise_linear_trend(self, t):
...
+
+        # floor or lowest point of logistic growth trend
+        if self.config_trend.trend_floor_user:
+            self.set_trend_floor(dataset)
+        else:
+            floor_quantile = torch.Tensor(torch.kthvalue(dataset.targets.squeeze(), 
+                                                         int(dataset.targets.shape[0] * self.trend_floor_init_quantile)))[0]
+            self.trend_floor = nn.Parameter(floor_quantile)
+
+    def set_trend_cap(self, dataset):
```
Comment (ourownstory) [inline neuralprophet/time_net.py:401]: can we run these checks before calling model?

**Review by ourownstory — CHANGES_REQUESTED** [review]

**Comment by schlinge2000** [comment]
how is the status of this work? are you going to merge it?

**Comment by Riley16** [comment]
When is the publication date? I won't be able to get to it this weekend.
Maybe next weekend.

**Pull request closed without merging**